### PR TITLE
fixed search block front-end placeholder issue.

### LINKED ISSUE #45837: Search placeholder not displaying on front-end
### Description

While working with the Twenty Twenty-Three theme, I have checked that the Search block is look weired

The search block placeholder is displaying on back-end. But, its not working on front-end and not displaying.

Here, I have attached its screenshots.

### Step-by-step reproduction instructions

1. Install latest wordpress(6.1.1)
2. Create new page and select search gutenberg block and publish that page.

### Screenshots, screen recording, code snippet

front-end
back-end

### Environment info

- WordPress Version 6.1.1
- Mac M1 OS 13.0
- Google chrome Version 107.0.5304.110 (Official Build) (arm64)

### Please confirm that you have searched existing issues in the repo.

No

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes, No

### PR DESCRIPTION
**Pull request by viralsampat-multidots**
## What?
The PR is added placeholder text into the search block for front-end view.

## Why?
I have checked search block and found that the placeholder text was displaying on editor side, But somehow it not displaying on front-end side. So, I have added placeholder text into the attributes array added it for front-end.

## How?
First, I have checked its code and found that attribute array key "Placeholder" is blank. So, I have added if condition and then added value for array key "Placeholder".

## Testing Instructions
1. Open a Post or Page.
2. Insert a Search Block.
3. Publish a Post or Page.

## Screenshots or scree
after_added_custom_placeholder

Closed: #45837

### CODE REVIEW

File: packages/block-library/src/search/index.php [hunk 1]
```diff
@@ -64,6 +64,11 @@ function render_block_core_search( $attributes ) {
 		);
 	}
 
+	// Check array attributes place holder value isset or not
+	if( empty( $attributes['placeholder'] ) ) {
+		$attributes['placeholder'] = 'Optional placeholder…';
```
Comment (alexstine) [inline packages/block-library/src/search/index.php]:
```suggestion
		$attributes['placeholder'] = __( 'Search here…' );
```

**Review by alexstine — COMMENTED** [review]
@viralsampat-multidots Thanks for the PR! Just one small suggestion.

**Comment by skorasaurus** [comment]
Hi, 

Thank you for your effort. At first impression, this would appear to be a logical and easy fix although I am skeptical whether WP should implement a placeholder (on the front end or the backend) in a search box. 

There are few posts from respectable accessibility and usability outlets that discourage their use. 
https://www.deque.com/blog/accessible-forms-the-problem-with-placeholders/ and https://digital.gov/2014/11/24/placeholder-text-think-outside-the-box/

File: packages/block-library/src/search/index.php (hunk 1 above)
Comment (viralsampat-multidots) [reply packages/block-library/src/search/index.php]: Hello @alexstine 

Hope you are doing well.

As I checked still this issue is open. So, Can I have to make this change and add another PR for review?

Please review it and let me know your feedback. Then I will work accordingly.

Thanks,

**Comment by alexstine** [comment]
I'd rather not have the placeholder at all. However, if it exists on the back-end, makes little sense to not include it on the front-end or remove it from the back-end.

**Comment by viralsampat-multidots** [comment]
> I'd rather not have the placeholder at all. However, if it exists on the back-end, makes little sense to not include it on the front-end or remove it from the back-end.

Hello @alexstine

I know the placeholder exists on the back end, makes little sense to not include it on the front end. But for accessibility and user interaction purposes, we can add it on the front end as well.

I have made the above-mentioned change and I have removed placeholder text from the back end as well.

Please let me know still need to make any changes or share your feedback for the same.

Thanks,

**Review by alexstine — APPROVED** [review]
LGTM.

**Comment by alexstine** [comment]
@viralsampat-multidots While this PR looks fine I think, probably need to remove this from the back-end.

**Comment by viralsampat-multidots** [comment]
Hello Team,

Here, I am seeing one error message like **"This branch has conflicts that must be resolved"**, So please let me know if I need to do anything to move further.

Please share your valuable feedback.

Thanks,

**Comment by aristath** [comment]
> Here, I am seeing one error message like "This branch has conflicts that must be resolved", So please let me know if I need to do anything to move further.

Hey there!
You will need to rebase the PR (or merge `trunk` in your branch, whichever is easier for you) and resolve the conflicts.

**Comment by alexstine** [comment]
@aristath How does this one move forward? I am actually unsure anything can happen here if this placeholder field on the back-end is currently being used. It is not displayed on the front-end so in theory it could be eliminated but this feels more for project leadership to decide. I am not 100% against placeholder text but I would prefer not introducing a feature that users will misuse.

It does appear that this is somehow being used though.

https://github.com/WordPress/wporg-documentation-2022/blob/trunk/source/wp-content/themes/wporg-documentation-2022/parts/search.html

If you check the front-end of the site, placeholder text is there.

https://wordpress.org/documentation/article/wordpress-block-editor/

I think we might be too late to fully eliminate placeholder text.

Thanks.

**Comment by viralsampat-multidots** [comment]
@alexstine I think that we have two options for now.
1) We can remove placeholder text completely from both places ( Front-end & Backend )
2) We can add placeholder text in both places( Front-end & Backend ).

What do you guys think? Please share your feedback for the same.

Thanks,

**Comment by skorasaurus** [comment]
Sorry for the delay @viralsampat-multidots; Thank you again for your effort. 

Personally, I would not have the use of placeholders although the degree of negative impact when they're used varies (based on color contrast, length of the placeholder text and of the text both etc). 
 
(A couple more articles that discourage the use of placeholders) 

https://design-system.service.gov.uk/components/text-input/

https://www.smashingmagazine.com/2018/06/placeholder-attribute/

How to proceed: 
the issue itself was closed by @carolinan a couple months ago so I would defer to her (as someone who's far more involved with the team).

**Pull request closed without merging**